server/models/attendanceSettingsModel.js

```diff
@@ -0,0 +1,58 @@
+import mongoose from 'mongoose'
+import AppError from "../AppError";
+import {userHasRole} from "../utils";
+import * as SC from "../serverconstants";
+import * as EC from "../errorcodes";
+import {isAuthenticated, isAdmin, isSuperAdmin, hasRole} from "../utils"
+
+mongoose.Promise = global.Promise
+
+let attendanceSettingsSchema = mongoose.Schema({
+
+    minFullDayHours:{type:Number, default:7},
+    minHalfDayHours:{type:Number, default:4},
+    dayStartTime:{type:String, default:"10:00"},
+    dayEndTime:{type:String, default:"22:00"}
+
+})
+
+/*I am hoping that default add / update / delete of super will work here */
+
+attendanceSettingsSchema.statics.getAdminAttendanceSettings = async (loggedInUser) => {
+
+    if (isAdmin || isSuperAdmim) {
+        // Only admin and super admin can see holidays
+        let settings =  await AttendanceSettingsModel.find().exec()
+        console.log("getAdminAttendanceSettings - ",settings);
+        if(!settings || settings.length == 0)
+        {
+            /**Default settings creation */
+            settings = new AttendanceSettingsModel();
```

Comment:
This code only works because there are default properties. In case there are no such properties an empty object would have been saved. Try to use new AttendanceSettingsModel({minFullDayHours:7, .....}). Also this is like a initial setup instruction and hence can become part of setupdata.js

Reply:
Yes I knew that and that's why i have used just with new. I also thought  where to put but can't find so did it like that. Good idea to put in setupdata.js